lib/webapi/webhandler.go

```diff
@@ -479,7 +479,8 @@ func (h *WebHandler) needsLogin(handle webHandle) httprouter.Handle {
 			if !trace.IsAccessDenied(err) {
 				log.Error(trace.DebugReport(err))
 			}
-			http.Redirect(w, r, "/web/login", http.StatusFound)
+
+			http.Redirect(w, r, "/web/login?redirect_uri="+r.URL.Path, http.StatusFound)
```

Comment:
Do you need to URL-encode `r.URL.Path`?

Reply:
Not in this case because we do not pass or take any information from this request.